modules/library/render/src/main/java/org/geotools/renderer/label/LabelCacheImpl.java

```diff
@@ -512,13 +513,19 @@ else if (((geom instanceof LineString) && !(geom instanceof LinearRing))
                          painted = paintLineLabelsWithLetterConflict(painter, tempTransform, displayArea, glyphs);
                      else
                          painted = paintLineLabels(painter, tempTransform, displayArea, glyphs);
-                     if (!painted){
+                     if (!painted) {
                          nonPaintedLineLabels++;
-                     } else paintedLineLabels++;
+                     } else {
+                         paintedLineLabels++;
+                     }
+                } else if (geom instanceof Polygon || geom instanceof MultiPolygon
+                        || geom instanceof LinearRing) {
+                    if(labelItem.getTextStyle().isPointPlacement() && !labelItem.isFollowLineEnabled()) {
+                        paintPolygonLabel(painter, tempTransform, displayArea, glyphs);
+                    } else {
+                        paintPolygonBorder(painter, tempTransform, displayArea, glyphs);
+                    }
```

Comment:
I would add a comment clarifying that the first if branch paints the polygon in the provided position and the second branch paints the label following the polygon border.